apps/functions/notify-subscribers/notify-subscribers/src/util.js

```diff
@@ -0,0 +1,53 @@
+import crypto from 'crypto';
+import { loadConfig } from './config.js';
+
+/**
+ * Returns a promise that waits for the given time before resolving.
+ *
+ * @param {number} ms
+ * @returns {Promise<void>}
+ */
+export function sleep(ms) {
+	return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+/**
+ * Get a project link - for the front office
+ *
+ * @param {string} caseReference
+ * @returns {string}
+ */
+export function projectLink(caseReference) {
+	const config = loadConfig();
+	return `${config.FRONT_OFFICE_URL}/projects/${caseReference}`;
+}
+
+/**
+ * Generate an unsubscribe link for the case reference + email address - this links to the front office
+ *
+ * @param {string} caseReference
+ * @param {string} emailAddress
+ * @returns {string}
+ */
+export function unsubscribeLink(caseReference, emailAddress) {
+	const config = loadConfig();
+	const email = encrypt(emailAddress);
+	return `${config.FRONT_OFFICE_URL}/projects/${caseReference}/get-updates/unsubscribe-confirm?email=${email};`;
+}
+
+/**
+ * Encrypt a value, used for unsubscribe links
+ *
+ * @see https://github.com/Planning-Inspectorate/applications-service/blob/main/packages/applications-service-api/src/lib/crypto.js
+ *
+ * @param {string} value
+ * @returns {string}
+ */
+export function encrypt(value) {
```

Comment:
Can we an Azure encryption service for this, so we don't have to manage encryption keys? Also if we really do need to use our own key, we should put it in key vault.

Reply:
Possibly, this is mirroring a Front Office function, since they are the ones decrypting. Wel share a key via key vault.

Reply:
Makes sense, sounds like a refactor for another time!